test/test_tournament.js

```diff
@@ -0,0 +1,56 @@
+let Tournament = artifacts.require('../contracts/Tournament');
+
+
+contract('Tournament', accounts => {
```

Comment:
I would like to propose to follow [arrange-act-assert](http://wiki.c2.com/?ArrangeActAssert) pattern of writing tests. It is a derivative of the [given-when-then](https://martinfowler.com/bliki/GivenWhenThen.html) approach. The main purpose of introducing it is to make tests easier to read. Every test should be split into 3 sections, where the act section should contain one and only one action.

Reply:
Actually I was wondering how given-when-then approach can be implemented here because before each of the batch of tests we are using 
```   
beforeEach(async () => {
        contract = await Tournament.new(
            id,
            buyIn,
        );
        participant = accounts[0];
    });
```

Reply:
I would just use `// act` and `// assert` in tests without the `// arrange` part

Reply:
For example
```
        it('user has sufficient buyIn amount', async () => {
            // act
            await contract.participate({
                value: buyIn,
                from: participant
            });

            // assert
            let isParticipantSaved = await contract.participants.call(participant);
            expect(isParticipantSaved).to.equal(true);
        });
```